components/Navbar/NavIconButton.tsx

```diff
@@ -0,0 +1,41 @@
+import { ButtonHTMLAttributes } from "react";
+
+import Bag from "./bagIcon.svg"; // in the final version it should be imported from ui-kit package
+import Close from "./closeIocn.svg"; // in the final version it should be imported from ui-kit package
+import MenuIcon from "./menuIcon.svg"; // in the final version it should be imported from ui-kit package
+import styles from "./Navbar.module.css"; // in the final version it should be imported from ui-kit package
+import Spyglass from "./spyglassIcon.svg"; // in the final version it should be imported from ui-kit package
+import User from "./userIcon.svg"; // in the final version it should be imported from ui-kit package
+
+interface NavIconButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  icon: "user" | "bag" | "spyglass" | "menu" | "close";
+  counter?: number;
+}
+
+const switchIcon = (icon: NavIconButtonProps["icon"]) => {
```

Comment:
I think we could name this function a bit better. One thing is that whatever place uses this function doesn't need to be aware of how it's implemented - I'd rather call it `getIcon` and I think the name `icon` as a prop where it's actually `iconName` is a bit of a misleading name, since by the definition here (considering the name of this function) you kind of return an `icon` as an icon component, but also (according to current naming of the prop) pass an `icon` as a prop - this time as an icon name.